etcd/src/main/scala/io/buoyant/etcd/Key.scala

```diff
@@ -0,0 +1,278 @@
+package io.buoyant.etcd
+
+import com.twitter.finagle.{Path, Service}
+import com.twitter.finagle.http._
+import com.twitter.io.Buf
+import com.twitter.util._
+import scala.collection.JavaConverters._
+
+case class BackoffsExhausted(key: Path, throwable: Throwable)
+  extends Exception(key.show, throwable)
+
+class Key(key: Path, client: Service[Request, Response]) {
+
+  import Etcd._
+
+  def path: Path = key
+
+  def key(path: Path): Key = new Key(key ++ path, client)
+  def key(name: String): Key = key(Path.read(name))
+
+  private[this] lazy val uriPath = keysPrefixPath ++ key
+
+  private[this] def boolParam(cond: Boolean) = if (cond) Some("true") else None
```

Comment:
these are each used in one method respectively.  I might move them inside or nearby those methods for readability.